src/Message/PurchaseRequest.php

```diff
@@ -70,11 +70,14 @@ private function sign($params, $api_key) {
 
 	public function sendData($data)
     {
+		// FIXME: You're not actually sending any data here.
```

Comment:
The reason that I raised this is because, for most gateways, you do actually send data in the `purchase()` call before making the redirect. The data that you send usually includes the returnUrl and perhaps cancelUrl, the amount, currency, and other transaction data. It's of course possible to send all of that in the GET that is the redirect but most gateways eschew that because it's possible for the customer to "fiddle" with that data before it arrives at the gateway, ending up with the customer perhaps paying a different amount to what was intended.

The response to that `purchase()` request usually contains a unique URL to redirect the customer to in order to make the payment.

See the PayPal_Rest gateway for a documented example of how this is done with PayPal.  Steps are (1) send a `purchase()` request to the gateway, (2) redirect the customer to the gateway, (3) receive the customer back on the returnUrl or the cancelUrl.  For security reasons I would be suspicious of any gateway that did not implement this type of flow, and would be looking elsewhere to get my payments processed.